Tools/_sunrise/changelog/actions_changelogs_since_last_run.py

```
@@ -422,6 +472,75 @@ def get_past_runs(sess: requests.Session, current_run: Any, page: int = 1) -> An
+        if match is not None and match.group("target").strip() == target_id:
+            matching.append(run)
+    if not matching:
+        raise RuntimeError(f"Не найден первый запуск публикации цели {target_id}")
+    return min(matching, key=lambda run: run["created_at"])
+
+
+def get_source_release_before_attempt(
+    sess: requests.Session,
+    github_repository: str,
+    source_run_id: str,
+    first_attempt: Any,
```

Ruff (0.16.2)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `first_attempt`

(ANN401)